File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

def DFS(v):
  global cnt
  visited[v] = 1

  for i in graph[v]:
    if visited[i] == 0:
      DFS(i)
      cnt += 1

def BFS(v):
  global cnt
  visited[v] = 1
  queue = [v]

  while queue:
    for i in graph[queue.pop()]:
      if visited[i] == 0:
        visited[i] = 1
        queue.insert(0, i)
        cnt += 1

n = int(sys.stdin.readline())
t = int(sys.stdin.readline())
graph = [[] for i in range(n + 1)]
cnt = 0

for i in range(t):
  a, b = map(int, sys.stdin.readline().split())
  graph[a].append(b)
  graph[b].append(a)

visited = [0] * (n + 1)
DFS(1)
# BFS(1)
sys.stdout.write(str(cnt) + "\n")

File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <iostream>
#include <vector>
using namespace std;

vector<int> graph[101];
bool visited[101];
int cnt = 0;

void DFS(int v) {
  visited[v] = true;

  for (int i = 0; i < graph[v].size(); i++) {
    int j = graph[v][i];

    if (visited[j] == false) {
      visited[j] = true;
      cnt++;
      DFS(j);
    }
  }

  return;
}

int main() {
  ios::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);

  int n, t;
  cin >> n >> t;

  for (int i = 0; i < t; i++) {
    int a, b;
    cin >> a >> b;

    graph[a].push_back(b);
    graph[b].push_back(a);
  }

  DFS(1);
  cout << cnt << "\n";

  return 0;
}